mlflow/utils/search_logged_model_utils.py

```diff
@@ -0,0 +1,142 @@
+import re
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional, Union
+
+import sqlalchemy
+import sqlparse
+
+from mlflow.exceptions import MlflowException
+from mlflow.store.tracking.dbmodels.models import SqlLoggedModel
+
+
+class EntityType(Enum):
+    ATTRIBUTE = "attributes"
+    METRIC = "metrics"
+    PARAM = "params"
+    TAG = "tags"
+
+    @classmethod
+    def from_str(cls, s: str) -> "EntityType":
+        if s == "attributes":
+            return cls.ATTRIBUTE
+        if s == "metrics":
+            return cls.METRIC
+        if s == "params":
+            return cls.PARAM
+        if s == "tags":
+            return cls.TAG
+
+        raise MlflowException.invalid_parameter_value(
+            f"Invalid entity type: {s!r}. Expected one of {[e.value for e in cls]}."
+        )
+
+
+@dataclass
+class Entity:
+    type: EntityType
+    key: str
+
+    IDENTIFIER_RE = re.compile(r"^([a-z]+)\.(.+)$")
+
+    def __repr__(self) -> str:
+        return f"{self.type.value}.{self.key}"
+
+    @classmethod
+    def from_str(self, s: str) -> "Entity":
+        if m := Entity.IDENTIFIER_RE.match(s):
+            return Entity(
+                type=EntityType.from_str(m.group(1)),
+                key=m.group(2).strip("`"),
+            )
+        return Entity(type=EntityType.ATTRIBUTE, key=SqlLoggedModel.ALIASES.get(s, s).strip("`"))
+
+    def is_numeric(self) -> bool:
+        """
+        Does this entity represent a numeric column?
+        """
+        return self.type == EntityType.METRIC or (
+            self.type == EntityType.ATTRIBUTE and SqlLoggedModel.is_numeric(self.key)
+        )
+
+    def validate_op(self, op: str) -> None:
+        numeric_ops = ("<", "<=", ">", ">=", "=", "!=")
+        string_ops = ("=", "!=", "LIKE", "ILIKE")
+        ops = numeric_ops if self.is_numeric() else string_ops
+        if op not in ops:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid comparison operator for {self}: {op!r}. Expected one of {string_ops}."
+            )
+
+
+@dataclass
+class Comparison:
+    left: Entity
+    op: str
+    right: Union[str, float]
```

Comment:
Could we update the field to be `key` `op` `value`? Feel like the name is not straightforward when used inside _apply_filter_string_search_logged_models

Reply:
Sounds good, will update it.